contracts/governancestaking/GovarnanceStaking.sol

```diff
@@ -0,0 +1,169 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.7.0;
+
+
+import "../Interfaces.sol";
+
+import "../DAOStackInterfaces.sol";
+import "../utils/NameService.sol";
+import "../utils/Pausable.sol";
+import "../utils/DAOContract.sol";
+import "./GovernanceStakingToken.sol";
+
+/**
+ * @title Staking contract that donates earned interest to the DAO
+ * allowing stakers to deposit Tokens
+ * or withdraw their stake in Tokens
+ * the contracts buy intrest tokens and can transfer the daily interest to the  DAO
+ */
+contract GovernanceStaking is Pausable, GovernanceStakingToken {
```

Comment:
why do we need a new BaseGovernanceShareField and GovernanceStakingToken? why cant we use the basesharefield etc?

Reply:
Because BaseGovernanceShareField logic differs in many ways than BaseShareField